### PR TITLE
refactor: replace archiver with @zip.js/zip.js

### PR DESCRIPTION
**Pull request by ctlai95**
## Problem

The `archiver` package uses `fs` which is not available in the browser. Running `npm run testWeb` results in the following error:

```
TypeError: Cannot read properties of undefined (reading 'native')
```

## Solution

- Replace `archiver` with [`@zip.js/zip.js`](https://gildas-lormeau.github.io/zip.js/) which has browser support.

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by justinmk3** [comment]
> `test macOS (16.x, minimum)` fails with `ReferenceError: TransformStream is not defined`

The "minimum" is testing a very old version of vscode, which has an older nodejs. https://github.com/aws/aws-toolkit-vscode/blob/ff3804f7e13078a88653a4fae77446ee8f461e37/packages/core/package.json#L12

I think we can and should bump the minimum to a vscode version that has node 16.

File: packages/toolkit/package.json
[the remark this answers is not in the document]
Comment (ctlai95) [reply packages/toolkit/package.json:1]: ah yes, let me revert it

File: packages/core/src/shared/utilities/zipStream.ts [hunk 1]
```diff
@@ -3,69 +3,135 @@
...
-import { getLogger } from '../logger'
+import { readFileAsString } from '../filesystemUtilities'
+// Use require instead of import since this package doesn't support commonjs
+const { ZipWriter, TextReader } = require('@zip.js/zip.js')
+import { getLogger } from '../logger/logger'
 
 export interface ZipStreamResult {
     sizeInBytes: number
-    md5: string
+    hash: string
```
Comment (pbalogh) [inline packages/core/src/shared/utilities/zipStream.ts:15]: Nice!

**Review by pbalogh — APPROVED** [review]

**Comment by justinmk3** [comment]
@ctlai95 vscode min version is now 1.83 : https://github.com/aws/aws-toolkit-vscode/pull/5704

**Comment by justinmk3** [comment]
Update: we will wait 1 more week before increasing min vscode to 1.83, see https://github.com/aws/aws-toolkit-vscode/pull/5749

**Comment by justinmk3** [comment]
> Update: we will wait 1 more week before increasing min vscode to 1.83, see #5749

vscode min version is now 1.83: https://github.com/aws/aws-toolkit-vscode/pull/5771

**Comment by ctlai95** [comment]
@justinmk3 I can remove `archiver` but I think we need to keep `stream-buffers`

**Pull request merged**